app/src/main/java/com/example/pix/home/fragments/HomeFragment.java

```diff
@@ -122,13 +124,30 @@ public boolean onQueryTextChange(String s) {
         colors.add(Color.GREEN);
 
 
+        ImageView profile = view.findViewById(R.id.home_profile_icon);
+        TextView pix = view.findViewById(R.id.tv_score);
         ViewPager viewPager = view.findViewById(R.id.vpPager);
         // Link the colors to each page
+        LinearLayout header = view.findViewById(R.id.header);
+        Drawable headerBackground = header.getBackground();
+        headerBackground.setAlpha(0);
         viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
             @Override
             public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
                 pagerTabStrip.setTabIndicatorColor(colors.get(position));
-                System.out.println(positionOffset);
+                // Only when not 0 because for some reason, when we are exactly at ComposeFragment(Index 1),
+                // The offset jumps to 0 from 0.9999
+                if (positionOffset != 0) {
+                    /*  When we scroll the PagerView, use the offset(The fraction of what index page we are on:
+                            Example: ChatsFragment is 0, ComposeFragment is 1, and in between them is 0.5
+                        We scale this index up to 255 and assign it to our header's background and the opposite
+                        as the icon tints. */
+                    int scaled = (int) (positionOffset * 255);
+                    headerBackground.setAlpha(255 - scaled);
+                    profile.setColorFilter(Color.argb(255, scaled, scaled, scaled));
+                    svChats.setColorFilter(Color.argb(255, scaled, scaled, scaled));
+                    pix.setTextColor(Color.argb(255, scaled, scaled, scaled));
```

Comment:
What is 255 here?

Reply:
The alpha value is out of 255. I will make it clear by making it a constant.